File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result;
        unordered_map<string,vector<string>> umap;
        for(auto i:strs){
            string s=i;
            sort(i.begin(),i.end());
            umap[i].push_back(s);
        }
        for(auto it:umap){
            result.push_back(it.second);
        }
        return result;
    }
};

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>> umap=new HashMap<>();
        List<List<String>> result=new ArrayList<>();

        for(String s:strs){
            char ch[]=s.toCharArray();
            Arrays.sort(ch);
            String sortedarray=new String(ch);
            if(!umap.containsKey(sortedarray)){
                umap.put(sortedarray,new ArrayList<>());
            }
            umap.get(sortedarray).add(s);
        }

        for(List<String> value:umap.values()){
            result.add(value);
        }
        return result;
    }
}